src/js/formats/helpers/buildCarousel.js

```diff
@@ -1,100 +1,73 @@
-    function setCurrent() {
-        q(`#carousel-slide-${currentSlide}`).forEach((el) => {
-            el.classList.add('js-current');
-            el.classList.add('explainer__carousel-slide--current');
-        });
         q(`#carousel-indicator-${currentSlide}`).forEach((el) => {
             el.children[0].classList.add('explainer__carousel-indicator--active');
             el.children[0].classList.add('js-active');
             el.children[0].classList.remove('explainer__carousel-indicator--passive');
         });
     }
 
-    function isFirstSlide() {
-        return (currentSlide <= 0);
-    }
-
-    function isLastSlide() {
-        return (currentSlide >= slideCount - 1);
-    }
-
     function setControlsVisibility() {
         q('.js-prev').forEach((el) => {
-            if (isFirstSlide()) {
+            if (slider.isFirstSlide()) {
                 el.classList.add('u-hidden');
             } else {
                 el.classList.remove('u-hidden');
             }
         });
         q('.js-next').forEach((el) => {
-            if (isLastSlide()) {
+            if (slider.isLastSlide()) {
                 el.classList.add('u-hidden');
             } else {
                 el.classList.remove('u-hidden');
             }
         });
     }
 
-    function updateSlides() {
-        clearCurrent();
-        setCurrent();
+    function updateControls() {
+        updateIndicator(slider.getCurrentSlide());
         setControlsVisibility();
     }
 
-    function showPreviousSlide() {
-        currentSlide -= 1;
-        updateSlides();
-    }
-
-    function showNextSlide() {
-        currentSlide += 1;
-        updateSlides();
-    }
-
-    setControlsVisibility();
-
-    q('.js-prev').forEach((el) => {
-        el.addEventListener('click', () => {
-            if (isFirstSlide()) {
-                return;
-            }
-            showPreviousSlide();
+    function bindEventHandlers() {
+        q('.js-prev').forEach((el) => {
+            el.addEventListener('click', () => {
+                slider.slideLeft();
+                updateControls();
+            });
         });
-    });
 
-    q('.js-next').forEach((el) => {
-        el.addEventListener('click', () => {
-            if (isLastSlide()) {
-                return;
-            }
-            showNextSlide();
+        q('.js-next').forEach((el) => {
+            el.addEventListener('click', () => {
+                slider.slideRight();
+                updateControls();
+            });
         });
-    });
 
-    q('.js-indicator').forEach((el) => {
-        el.addEventListener('click', () => {
-            const elementId = el.id || el.parentNode.id;
-            const newSlideIndex = elementId.split(/carousel-indicator-/)[1];
+        q('.js-indicator').forEach((el) => {
+            el.addEventListener('click', () => {
+                const elementId = el.id || el.parentNode.id;
+                const newSlideIndex = elementId.split(/carousel-indicator-/)[1];
```

Comment:
I'd prefer to use a `data-` attribute rather than the id to link indicators to slides